src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -1,15 +1,246 @@
+open Async
+open Core
+
 module Node = struct
+  (* a pod name, e.g., test-block-producer-1 *)
   type t = string
 
+  let to_string t = t
+
+  let full_name_opt : string option ref = ref None
+
+  let set_full_name name = full_name_opt := Some name
+
+  let get_full_name () = !full_name_opt
+
+  let set_name_from_network ~logger ~namespace ~node =
```

Comment:
We can avoid this by pushing this logic down into the shell. For instance, whenever we need to `exec_command node cmd`, we can shell out the command ` kubectl -n <namespace> exec -it $(kubectl get pod -n <namespace> -l app=<pod-prefix-name> -o name) -- bash`. Doing this will also help enforce that we always need to lookup the full pod's name (since the full pod name can be changed if the pod is evicted).